plugins/woocommerce/tests/e2e-pw/utils/themes.js

```diff
@@ -1,18 +1,48 @@
-const { exec } = require( 'node:child_process' );
+const { encodeCredentials } = require( '../utils/plugin-utils' );
+const https = require( 'https' );
+const http = require( 'http' );
+const { admin } = require( '../test-data/data' );
+const { BASE_URL } = process.env;
 
 export const DEFAULT_THEME = 'twentytwentythree';
 
 export const activateTheme = ( themeName ) => {
 	return new Promise( ( resolve, reject ) => {
-		const command = `wp-env run tests-cli wp theme install ${ themeName } --activate`;
+		const url = new URL( BASE_URL );
+		const isHttp = url.protocol === 'http:';
+		const requestModule = isHttp ? http : https;
 
-		exec( command, ( error, stdout ) => {
-			if ( error ) {
-				console.error( `Error executing command: ${ error }` );
-				return reject( error );
-			}
+		const options = {
+			hostname: url.hostname,
+			port: url.port || ( isHttp ? 80 : 443 ),
+			path: '/wp-json/e2e-theme/activate',
+			method: 'POST',
+			headers: {
+				Authorization: `Basic ${ encodeCredentials(
+					admin.username,
+					admin.password
+				) }`,
+				'Content-Type': 'application/json',
+			},
+		};
 
-			resolve( stdout );
+		const req = requestModule.request( options, ( res ) => {
+			let data = '';
+
+			res.on( 'data', ( chunk ) => {
+				data += chunk;
+			} );
+
+			res.on( 'end', () => {
+				resolve( JSON.parse( data ).message );
+			} );
 		} );
+
+		req.on( 'error', ( error ) => {
+			reject( error );
+		} );
+
+		req.write( JSON.stringify( { theme_name: themeName } ) );
+		req.end();
```

Comment:
Why is this so complicated? It should only be a simple API call like so many others in these tests. 
- The Promise part is most probably left over from the previous cli execution.
- Why do we need to set a port? Have you tested this with an external env? I doubt it will work.

Reply:
It is not complicated that much, just properly written to use Promise and not callback pattern. Would you use callback instead?

The port is here to help with any custom ones or explicit ones in the future.

It works nicely across localhost, Pressable and WPCOM.

Reply:
If the port is required it's already part of the baseURL defined in the config.

`port: url.port || ( isHttp ? 80 : 443 )` => this will add a port if there is none specified. Why do you need to add the port for external site urls?

All the other usage of `req` is useless.

How about:

```
const { request } = require( '@playwright/test' );
const { encodeCredentials } = require( '../utils/plugin-utils' );
const { admin } = require( '../test-data/data' );

export const DEFAULT_THEME = 'twentytwentythree';

export const activateTheme = async ( baseURL, theme ) => {
	const requestContext = await request.newContext( {
		baseURL,
		extraHTTPHeaders: {
			Authorization: `Basic ${ encodeCredentials(
				admin.username,
				admin.password
			) }`,
			cookie: '',
		},
	} );

	return await requestContext
		.post( '/wp-json/e2e-theme/activate', {
			theme_name: theme,
		} )
		.then( ( response ) => {
			return response.json();
		} );
};
```

Shorter, cleaner and consistent to the rest of the utils.

Reply:
That looks nice, but the error is then: [apiRequestContext.post: Invalid URL](https://d.pr/i/COmj5w)

Reply:
That's because the function definition changed, and you'll of course need to update all the function calls to send the `baseURL` as a parameter. The previous definition only took 1 parameter (the theme), but now it takes 2.

Reply:
Ah, I see. That works now with a bit modified your code to post with `data`:
```
.post( '/wp-json/e2e-theme/activate', {
	data: {
		theme_name: theme,
	},
```

Reply:
Theme util updated with simplified version and updated all the relevant tests with `activateTheme` function. Tested across local, external envs and it worked fine.